src/codegen.cpp

```diff
@@ -4647,57 +4588,211 @@ static std::unique_ptr<Module> emit_function(jl_lambda_info_t *lam, jl_llvm_func
                 else {
                     inl_name = "macro expansion";
                 }
-                SP = dbuilder.createFunction(new_file,
-                                             inl_name + ";",
-                                             inl_name,
-                                             new_file,
-                                             0,
-                                             jl_di_func_sig,
-                                             false,
-                                             true,
-                                             0,
-                                             0,
-                                             true,
-                                             nullptr);
-                MDNode *inlinedAt = NULL;
+                if (ctx.debug_enabled) {
+                    SP = dbuilder.createFunction(new_file,
+                                                 std::string(inl_name) + ";",
+                                                 inl_name,
+                                                 new_file,
+                                                 0,
+                                                 jl_di_func_sig,
+                                                 false,
+                                                 true,
+                                                 0,
+                                                 0,
+                                                 true,
+                                                 nullptr);
+                    MDNode *inlinedAt = NULL;
 #ifdef LLVM37
-                inlinedAt = builder.getCurrentDebugLocation();
+                    inlinedAt = cur_prop.loc;
 #else
-                inlinedAt = builder.getCurrentDebugLocation().getAsMDNode(jl_LLVMContext);
+                    inlinedAt = cur_proc.loc.getAsMDNode(jl_LLVMContext);
```

Comment:
And I think Jameson secretly fixed this one in #18125 too https://github.com/JuliaLang/julia/blob/b7f3127940df7bf779d377865aee04490a5a6e0a/src/codegen.cpp#L4265 I guess I'll just wait for that one unless someone really want to use LLVM 3.3 / MSVC with unpatched master before #18125 is ready.